### PR TITLE
Create proxy command runner to toggle between platform mode worklow

### PR DESCRIPTION
**Pull request by msarvar**
Created a `PlatformModeFeatureRunner` that acts as a proxy for plan, apply and policy check command runners.
Created dummy plan and apply runners that only comment to github

### CODE REVIEW

File: server/events/command/policies/runner.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		vcsClient: vcsClient,
+	}
+}
+
+type Runner struct {
+	vcsClient vcs.Client
+}
+
+func (r *Runner) Run(ctx *command.Context, cmd *events.CommentCommand) {
+	if err := r.vcsClient.CreateComment(ctx.Pull.BaseRepo, ctx.Pull.Num, "I'm a platform mode approve_policies runner", command.ApprovePolicies.String()); err != nil {
```
Comment (nishkrishnan) [inline server/events/command/policies/runner.go:20]: This description lmao.

File: server/lyft/command/feature_runner.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	featureAllocator    feature.Allocator
+	platformModeEnabled bool
+	logger              logging.SimpleLogging
+}
+
+type featureRunnerFunc func(ctx *command.Context, cmd *events.CommentCommand)
+
+func (f featureRunnerFunc) Run(ctx *command.Context, cmd *events.CommentCommand) {
+	f(ctx, cmd)
+}
```
Comment (nishkrishnan) [inline server/lyft/command/feature_runner.go:35]: Why is this necessary vs. just an interface? All the methods seem to have the same signature?  Is there a reason we can't just use a decorator?

File: server/lyft/command/feature_runner.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+) featureRunnerFunc {
+	return func(ctx *command.Context, cmd *events.CommentCommand) {
+		// if platform mode is not enable run unallocatedRunner runner. No need
+		// to invoke feature allocator
+		if !r.platformModeEnabled {
+			unallocatedRunner.Run(ctx, cmd)
+			return
+		}
+
+		shouldAllocate, err := r.featureAllocator.ShouldAllocate(feature.PlatformMode, ctx.HeadRepo.FullName)
```
Comment (nishkrishnan) [inline server/lyft/command/feature_runner.go:51]: not in this PR but something we should do is emit stats within the feature allocator when there is an error.

File: server/lyft/command/feature_runner.go (hunk 2 above)
Comment (msarvar) [reply server/lyft/command/feature_runner.go:35]: This is to avoid instantiating a new decorator instance for each command. For instance in `server.go` I'm doing this:
```
	featuredRunner := lyftCommands.NewPlatformModeFeatureRunner(
		featureAllocator,
		userConfig.EnablePlatformMode,
		logger,
	)

	commentCommandRunnerByCmd := map[command.Name]comment.Runner{
		command.Plan: featuredRunner.Wrap(
			plan.NewRunner(vcsClient),
			planCommandRunner,
		),
		command.Apply: featuredRunner.Wrap(
			apply.NewRunner(vcsClient),
			applyCommandRunner,
		),
		command.ApprovePolicies: featuredRunner.Wrap(
			policies.NewRunner(vcsClient),
			approvePoliciesCommandRunner,
		)
	}
```
Instead of initializing a new runner for each plan. This is basically a decorator pattern as well, just more functional inspired. Based on [this](https://www.youtube.com/watch?v=xyDkyFjzFVc) talk. I also fixed the signature of the `Wrap` function to return `CommentCommandRunner` interface.

Comment (msarvar) [reply server/lyft/command/feature_runner.go]: I changed the code to use decorator struct.

**Review by nishkrishnan — APPROVED** [review]

**Review by nishkrishnan — APPROVED** [review]
